webapi/src/test/scala/org/knora/webapi/e2e/v2/ValuesRouteV2E2ESpec.scala

```diff
@@ -2701,6 +2701,44 @@ class ValuesRouteV2E2ESpec extends E2ESpec {
             savedTargetIri should ===(linkTargetIri)
         }
 
+        "update a link between two resources, changing only the comment" in {
```

Comment:
The link value that gets updated here was created in

https://github.com/dasch-swiss/knora-api/blob/b823a3f2d38552654c82236c636b7340b3a26325/webapi/src/test/scala/org/knora/webapi/e2e/v2/ValuesRouteV2E2ESpec.scala#L1697

without a comment. Then a comment gets added in "update a link between two resources, changing only the comment". Shouldn't we ideally have test cases for

- create a value with an initial comment that then gets changed
- create a value without a comment that is added

Reply:
Good idea, I'll do that.

Reply:
Ideally, you could also do that for `https://github.com/dasch-swiss/knora-api/pull/1582/files#diff-0cbc5ab548dbaeaf4be4a8807abb05b7`.

Maybe it would not hurt to do it for a non link value too. I guess these have a common template so testing one non linking value would be ok to begin with, right?

Reply:
Yes, it's only link values that are handled by separate templates.

Reply:
Great! 

Is this understanding correct?

> To create a new version of a value either the value itself has to be different (regardless of its comment being different from the previous version) or just its comment (this includes submitting a comment for a value that did not have any comment before).

Reply:
It should be correct. I'll add the additional tests you asked for to make sure. :)

Reply:
What about submitting the same value without a comment for a value that had a comment before?

Reply:
I am about to create validators for the GUI elements and they should disallow the submission of invalid combinations.

Reply:
That should remove the comment from the value.

Reply:
> That should remove the comment from the value.

Is there already a test for this? ;-)

Reply:
Will add a test for that.

Reply:
great!

Reply:
I added a lot of tests, see if this looks OK to you.